packages/core/src/icon/icon.ts

```diff
@@ -1,36 +1,116 @@
 import type { CSSResultGroup, PropertyValues, TemplateResult } from 'lit';
+import type { IconPrefix, IconStyle } from '@fortawesome/fontawesome-common-types';
+import type { IconDefinition, IconName } from '@fortawesome/fontawesome-svg-core';
```

Comment:
What I did in DNA was to copy the types I needed into DNA. That way I did not have a runtime dependency on font awesome. Perhaps we should do the same? As long as the types match, typescript is fine.